src/app/shared/components/icon-button/icon-button.component.ts

```diff
@@ -8,15 +8,15 @@ import { IconService } from 'src/app/core/abstractions/icon.service'
   templateUrl: './icon-button.component.html',
   styleUrls: ['./icon-button.component.scss'],
 })
-export class IconButtonComponent implements OnInit {
+export class IconButtonComponent implements OnInit, OnChanges {
   @Input()
   public iconButton: IconButton
```

Comment:
We should consider if IconButton is the correct name for the input and enum. It is not a button we input. It is a icon type.

Reply:
I guess it would make more sense to call the enum `ButtonIcon`?

Reply:
Or just `Icon` cause that is what it is

Reply:
I have renamed the enums to `Icon` and `IconSize`.